src/index.js

```diff
@@ -92,13 +92,20 @@ class CID {
     return this.buffer
   }
 
+  toV0String () {
+    if (this.version === 0) {
+      return mh.toB58String(this.multihash)
+    }
+
+    throw new Error(`Not supported on version: ${this.version}`)
+  }
+
   /* defaults to base58btc */
   toBaseEncodedString (base) {
     base = base || 'base58btc'
 
     switch (this.version) {
       case 0:
-        return mh.toB58String(this.multihash)
       case 1:
```

Comment:
**note:**, for IPFS it is super fine to receive a CIDV1 that is addressing content using `dag-pb`, if people need to use different bases because of domain names or other reasons, we can have a simple tool that does `cidv0tov1`, because after the first cid being passed to IPFS (the root hash of something), IPFS will (at least should, if not, it is a bug) be able to traverse the rest of the graph, independent of it being CIDV0 or CIDV1.